### PR TITLE
Updated for IDA 9.0

### PR DESCRIPTION
**Pull request by Nemirtingas**

### CODE REVIEW

File: keypatch.py [hunk 1]
```diff
@@ -310,33 +311,48 @@ def __init__(self, arch=None, mode=None):
     def get_hardware_mode():
         (arch, mode) = (None, None)
 
-        # heuristically detect hardware setup
-        info = idaapi.get_inf_structure()
+        try:
```
Comment (kabeor) [inline keypatch.py:314]: Plz change all `try except` to `idaapi.IDA_SDK_VERSION >= 900` like code, thanks!

**Comment by Nemirtingas** [comment]
@kabeor , done!

File: keypatch.py (hunk 1 above)
Comment (Nemirtingas) [reply keypatch.py]: @kabeor , done!

**Review by kabeor — APPROVED** [review]
Thank you! Merged.

**Pull request merged**